From 501aaaba17b29ca547b4ef1202be750bd21ea761 Mon Sep 17 00:00:00 2001
From: Vinu Palackal <vinup@tataelxsi.co.in>
Date: Thu, 5 Feb 2015 00:52:56 +0000
Subject: [PATCH 8/9] Revert "BPV-8919 : Adding the fixes in dibbler for fixing
 the segmentation faults Reason for change: Dibbler causing segmentation fault
 in master Test Procedure: Provision box in ipv6 mode and start dibbler client
 Risks: Low"

Builds are failing due to this check in

This reverts commit e10c955715a7a8febad0ea1c88557e1ba2019238.

Change-Id: Ie67955bcd93ae74be6c2a8423b3fe9022740a12f
---
 Port-linux/lowlevel-options-linux.c | 23 +++++------------------
 1 file changed, 5 insertions(+), 18 deletions(-)

diff --git a/Port-linux/lowlevel-options-linux.c b/Port-linux/lowlevel-options-linux.c
index 0407307..c79fdbf 100644
--- a/Port-linux/lowlevel-options-linux.c
+++ b/Port-linux/lowlevel-options-linux.c
@@ -479,18 +479,13 @@ int prefix_add(const char* ifname, int ifindex, const char* prefixPlain, int pre
     if (!f) {
 	/* unable to open, so this file is missing, let's create it */
 	f = fopen(RADVD_FILE, "w");
-	if (f == NULL) {
-		sprintf(errorMsg, "Unable to create %s file. \n", RADVD_FILE);
-	}
-	else {
-		fprintf(f, "#\n");
-		fprintf(f, "# Router Advertisement config file generated by Dibbler %s\n", DIBBLER_VERSION);
-		fprintf(f, "#\n");
-		fprintf(f, "\n");
-	}
+	fprintf(f, "#\n");
+	fprintf(f, "# Router Advertisement config file generated by Dibbler %s\n", DIBBLER_VERSION);
+	fprintf(f, "#\n");
+	fprintf(f, "\n");
     }
     if (!f) {
-	sprintf(errorMsg, "Unable to open %s file.\n", RADVD_FILE);
+	sprintf(errorMsg, "Unable to open %s file.", RADVD_FILE);
 	return LOWLEVEL_ERROR_FILE;
     }
     fseek(f, 0, SEEK_END);
@@ -547,15 +542,7 @@ int prefix_del(const char* ifname, int ifindex, const char* prefixPlain, int pre
     rename(RADVD_FILE,RADVD_FILE".old");
 
     f = fopen(RADVD_FILE".old","r");
-    if (f == NULL) {
-	sprintf(errorMsg, "Unable to open %s file. \n", RADVD_FILE".old");
-	return LOWLEVEL_ERROR_FILE;
-    }
     f2 = fopen(RADVD_FILE,"w"); 
-    if (f2 == NULL) {
-	sprintf(errorMsg, "Unable to open %s file. \n", RADVD_FILE);
-	return LOWLEVEL_ERROR_FILE;
-    }
 
     snprintf(buf2, 511, "### %s start ###\n", ifname);
     while (fgets(buf,511,f)) {
-- 
2.3.1

